frontend/dockerfile/docs/reference.md

```diff
@@ -1299,16 +1325,41 @@ FROM alpine
 ADD git@git.example.com:foo/bar.git /bar
 ```
 
-This Dockerfile can be built with `docker build --ssh` or `buildctl build --ssh`, e.g.,
+This Dockerfile can be built with `docker build --ssh`, for example:
 
 ```console
 $ docker build --ssh default
 ```
 
-```console
-$ buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=. --ssh default
+### Destination
+
+If the destination path begins with a forward slash, it's interpreted as an
+absolute path, and the source files are copied into the specified destination
+relative to the root of the current build stage.
+
+```dockerfile
+# create /abs/test.txt
+ADD test.txt /abs/
 ```
 
+Trailing slashes are significant. For example, `ADD test.txt /abs` creates a
+file at `/abs`, whereas `ADD test.txt /abs/` creates `/abs/test.txt`.
+
+If the destination path doesn't begin with a leading slash, it's interpreted as
+relative to the working directory of the build container.
+
+```dockerfile
+WORKDIR /usr/src/app
+# create /usr/src/app/rel/test.txt
+ADD test.txt rel/
+```
+
+If destination doesn't exist, it's created, along with all missing directories
+in its path.
+
+If the source is a file, and the destination doesn't end with a trailing slash,
+the source file will be written to the destination path as a file.
```

Comment:
Should somewhere here in `ADD` maybe reference `COPY` for the permissions bit?  (maybe not all that important either, since folks _just_ copying files should be using `COPY` anyways?)

Reply:
Added the permission bits here too, I don't mind repeating it.